src/signutil.cpp

```diff
@@ -0,0 +1,197 @@
+      ctx(),
+      output.data(),
+      &outputlen,
+      &pubkey,
+      SECP256K1_EC_UNCOMPRESSED);
+
+  if (!res) {
+    throw std::invalid_argument("publicKeyUncompress(): public key serialize failed");
+  }
+  return output;
+}
+
+static std::vector<uint8_t> publicKeyUncompressedToAsn1(Slice<uint8_t> publicKey) {
+  checkRange(publicKey.size(),
+             PUBLIC_KEY_UNCOMPRESSED_SIZE, PUBLIC_KEY_UNCOMPRESSED_SIZE);
+  std::vector<uint8_t> result{};
+  result.reserve(ASN1_PREFIX_PUBKEY_BYTES.size() + publicKey.size());
+  result.insert(
+      result.end(), ASN1_PREFIX_PUBKEY_BYTES.begin(), ASN1_PREFIX_PUBKEY_BYTES.end());
+  result.insert(result.end(), publicKey.begin(), publicKey.end());
+  return result;
+}
+
+static std::vector<uint8_t> publicKeyAsn1ToUncompressed(Slice<uint8_t> publicKey) {
+  checkRange(publicKey.size(), PUBLIC_KEY_ASN1_SIZE, PUBLIC_KEY_ASN1_SIZE);
+  std::vector<uint8_t> result{};
+  result.reserve(PUBLIC_KEY_UNCOMPRESSED_SIZE);
+  result.insert(result.end(),
+                publicKey.end() - PUBLIC_KEY_UNCOMPRESSED_SIZE,
+                publicKey.end());
+  return result;
+}
+
+PrivateKey privateKeyFromVbk(Slice<uint8_t> key) {
+  return getPrivateKeyFromAsn1(key);
+}
+
+PublicKey publicKeyFromVbk(Slice<uint8_t> key) {
+  if (key.size() == PUBLIC_KEY_ASN1_SIZE) {
+    if (key[23] != 0x04) {
+      throw std::invalid_argument(
+          "publicKeyFromVbk(): unknown public key format");
+    }
+
+    return publicKeyAsn1ToUncompressed(key);
+  }
+
+  if (key.size() == PUBLIC_KEY_UNCOMPRESSED_SIZE) {
+    if (key[0] != 0x04) {
+      throw std::invalid_argument(
+          "publicKeyFromVbk(): unknown public key format");
+    }
+    return std::vector<uint8_t>(key.begin(), key.end());
+  }
+
+  if (key.size() == PUBLIC_KEY_COMPRESSED_SIZE) {
+    if (key[0] != 0x02 && key[0] != 0x03) {
+      throw std::invalid_argument(
+          "publicKeyFromVbk(): invalid key format, expected 0x02 or 0x03 as "
+          "first byte");
+    }
+
+    return publicKeyUncompress(key);
+  }
+
+  throw std::invalid_argument("publicKeyFromVbk(): invalid public key");
+}
+
+std::vector<uint8_t> publicKeyToVbk(PublicKey key) {
+  return publicKeyUncompressedToAsn1(key);
+}
+
+PublicKey derivePublicKey(PrivateKey privateKey) {
+  checkRange(privateKey.size(), PRIVATE_KEY_SIZE, PRIVATE_KEY_SIZE);
+  Secp256k1Context ctx{secp256k1_context_create(SECP256K1_CONTEXT_SIGN)};
+  secp256k1_pubkey pubkey;
+  secp256k1_ec_pubkey_create(ctx(), &pubkey, privateKey.data());
+
+  size_t outputlen = PUBLIC_KEY_UNCOMPRESSED_SIZE;
+  std::vector<uint8_t> output(outputlen);
+  int res = secp256k1_ec_pubkey_serialize(
+      ctx(),
+      output.data(),
+      &outputlen,
+      &pubkey,
+      SECP256K1_EC_UNCOMPRESSED);
+  if (!res) {
+    throw std::invalid_argument(
+        "derivePublicKey(): public key serialize failed");
+  }
+  return output;
+}
+
+std::vector<uint8_t> veriBlockSign(Slice<uint8_t> message,
```

Comment:
```suggestion
std::vector<uint8_t> veriBlockSign(Slice<const uint8_t> message,
```
here and in other places

the difference is same as between
```
uint8_t* 
and
const uint8_t*
```